pyuvsim/antenna.py

```diff
@@ -71,14 +69,23 @@ def get_beam_jones(self, array, source_alt_az, frequency, reuse_spline=True,
         else:
             freq = np.array([frequency])
 
-        if array.beam_list[self.beam_id].data_normalization != 'peak':
-            array.beam_list[self.beam_id].peak_normalize()
+        beam = array.beam_list[self.beam_id]
+
+        if beam.data_normalization != 'peak':
+            beam.peak_normalize()
 
         if freq_interp_kind is not None:
-            array.beam_list[self.beam_id].freq_interp_kind = freq_interp_kind
+            beam.freq_interp_kind = freq_interp_kind
 
         if interpolation_function is not None:
-            array.beam_list[self.beam_id].interpolation_function = interpolation_function
+            beam.interpolation_function = interpolation_function
+
+        # If interpolation_function is unset, or the beam lacks the attribute
+        # (i.e., is analytic) then default to az_za_simple.
+        if getattr(beam, 'interpolation_function', 'analytic') is None:
```

Comment:
I find the comment a little confusing -- it doesn't default to anything for an analytic beam if I'm reading the code right. Which is fine, there's no interpolation to do for an analytic beam.

Reply:
I've rewritten the lines. I just wanted something that would assign a default to UVBeams when the interpolation_function isn't set but not do anything to analytic beams, and this was quick way to do it.